src/Solnet.Rpc/Builders/TransactionBuilder.cs

```diff
@@ -45,9 +45,11 @@ public TransactionBuilder()
         /// <summary>
         /// Serializes the message into a byte array.
         /// </summary>
-        private byte[] Serialize()
+        public byte[] Serialize()
         {
             byte[] signaturesLength = ShortVectorEncoding.EncodeLength(_signatures.Count);
+            if (_serializedMessage == null)

```

Comment:
can we add a parameter to possibly force either way?
rationale: given tx size constraints, this would allow one to figure optimal TX contents during runtime for variable-length instruction data